Source/Core/DolphinWX/InputConfigDiag.h

```diff
@@ -83,6 +84,37 @@ class PadSettingCheckBox : public PadSetting
 	ControllerEmu::ControlGroup::Setting* const setting;
 };
 
+class InputEventFilter : public wxEventFilter
+{
+public:
+	InputEventFilter()
+	{
+		wxEvtHandler::AddFilter(this);
+	}
+
+	~InputEventFilter()
+	{
+		wxEvtHandler::RemoveFilter(this);
+	}
+
+	int FilterEvent(wxEvent& event) override;
+
+	void BlockEvents(bool block) { m_block = block; }
+
+private:
+	static bool ShouldCatchEventType(wxEventType type)
+	{
+		// Can't use a switch... "expression must have a constant value"
```

Comment:
I'd say a switch over event types would be a reasonable thing to do, but
I'll remove it after getting some sleep (3:23 AM here).
Am 04.10.2015 03:12 schrieb "shuffle2" <notifications@github.com>:

> In Source/Core/DolphinWX/InputConfigDiag.h
> <https://github.com/dolphin-emu/dolphin/pull/3068#discussion_r41093348>:
>
> > +		wxEvtHandler::AddFilter(this);
> > +	}
> > +
> > +	~InputEventFilter()
> > +	{
> > +		wxEvtHandler::RemoveFilter(this);
> > +	}
> > +
> > +	int FilterEvent(wxEvent& event) override;
> > +
> > +	void BlockEvents(bool block) { m_block = block; }
> > +
> > +private:
> > +	static bool ShouldCatchEventType(wxEventType type)
> > +	{
> > +		// Can't use a switch... "expression must have a constant value"
>
> This comment suggests you've tried to do something strange, and it didn't
> work. Doesn't need a comment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dolphin-emu/dolphin/pull/3068/files#r41093348>.
>